e2etests/pkg/k8s/pods.go

```diff
@@ -32,3 +40,70 @@ func FRRK8sPods(cs clientset.Interface) ([]*corev1.Pod, error) {
 	}
 	return res, nil
 }
+
+func RestartFRRK8sPods(cs clientset.Interface) error {
+	pods, err := FRRK8sPods(cs)
+	if err != nil {
+		return err
+	}
+	oldNames := []string{}
+	for _, p := range pods {
+		oldNames = append(oldNames, p.Name)
+		err := cs.CoreV1().Pods(FRRK8sNamespace).Delete(context.Background(), p.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	return wait.PollUntilContextTimeout(context.Background(), 100*time.Millisecond,
+		1*time.Minute, false, func(context.Context) (bool, error) {
+			npods, err := FRRK8sPods(cs)
+			if err != nil {
+				return false, err
+			}
+			for _, p := range npods {
+				if slices.Contains(oldNames, p.Name) {
+					return false, nil
+				}
+				if !podIsRunningAndReady(p) {
+					return false, nil
+				}
+			}
+			return true, nil
+		})
+}
+
+func podIsRunningAndReady(pod *v1.Pod) bool {
+	if pod.Status.Phase != v1.PodRunning {
+		return false
+	}
+
+	for _, containerStatus := range pod.Status.ContainerStatuses {
+		if !containerStatus.Ready {
+			return false
+		}
+	}
+
+	return true
+}
+
+func FRRK8isDaemonSetReady(clientset clientset.Interface) (bool, error) {
+
+	// common labels in heml, kustomize
+	l := "app.kubernetes.io/component=frr-k8s,app.kubernetes.io/name=frr-k8s"
```

Comment:
A bit uncertain if worth to put as const and on top of a file

Reply:
let's use the local one, we are using it only here after all